### PR TITLE
Add Update Product API endpoint docs

### PR DESCRIPTION
**Pull request by OttilieWilliams**
Connects CorporateRewards/redstone#1135

### CODE REVIEW

**Comment by OttilieWilliams** [comment]
Links https://github.com/CorporateRewards/redstone/issues/1135

File: source/includes/api/products/update_product.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
+## Update product
+
+The update product API is available to update product information. This uses the same params as the create product API above.
+
+> Request:
+
+``` http
+PATCH /api/v2/products/{123} HTTP/1.1
```
Comment (rjpaskin) [inline source/includes/api/products/update_product.md]:
```suggestion
PATCH /api/v2/products/{product_id} HTTP/1.1
```

File: source/includes/api/products/update_product.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
+## Update product
+
+The update product API is available to update product information. This uses the same params as the create product API above.
```
Comment (rjpaskin) [inline source/includes/api/products/update_product.md]: Link to section for "same params as the create product API"?

**Review by rjpaskin — CHANGES_REQUESTED** [review]

File: source/includes/api/products/update_product.md [hunk 3]
```diff
@@ -1,11 +1,11 @@
 ## Update product
 
-The update product API is available to update product information. This uses the same params as the create product API above.
+The update product API is available to update product information. This uses the same params as the [Create product API](https://docs.gps.my-rewards.co.uk/#products-create-product) above.
```
Comment (rjpaskin) [inline source/includes/api/products/update_product.md]: Don't need the full URL here, just the anchor:
```suggestion
The update product API is available to update product information. This uses the same params as the [Create product API](#products-create-product) above.
```

**Review by rjpaskin — CHANGES_REQUESTED** [review]

**Review by rjpaskin — APPROVED** [review]

**Pull request merged**